packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1656,13 +1657,15 @@ class ArtifactUpdater {
   }
 
   /// Download bytes from [url], throwing non-200 responses as an exception.
-  Future<void> _download(Uri url, IOSink ioSink) async {
+  Future<void> _download(Uri url, File file) async {
     final HttpClientRequest request = await _httpClient.getUrl(url);
     final HttpClientResponse response = await request.close();
     if (response.statusCode != HttpStatus.ok) {
       throw Exception(response.statusCode);
     }
-    await response.forEach(ioSink.add);
+    await response.forEach((List<int> chunk) {
+      file.writeAsBytesSync(chunk, mode: FileMode.append);
```

Comment:
NB: This is going to open and close the file on every chunk, which might be significant since the cache is downloading large files.

Reply:
I think it's not worth it right now, but that would be a good plan if we notice that there's a problem with the change in this PR.

Reply:
ack